lib/cfa/matcher.rb

```diff
@@ -41,4 +75,14 @@ def value_match?(element, matcher)
       end
     end
   end
+
+  # An abstract class documenting the block argument to {Matcher#initialize}
+  class MatcherBinaryPredicate < Proc
```

Comment:
Benefit: clear documentation with hyperlinks in Yard, naming the concept
Cost: few bytes of code in memory

My conclusion is that the benefit is much bigger. Have I overlooked anything? Do you suggest an alternative?